api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ConceptTranslatorImpl.java

```diff
@@ -103,6 +103,11 @@ public Concept toOpenmrsType(@Nonnull CodeableConcept concept) {
 							    coding.getCode());
 							if (c.isPresent()) {
 								return c.get();
+							} else {
+								c = conceptService.getConceptWithAnyMappingInSource(conceptSource.get(), coding.getCode());
+								if (c.isPresent()) {
+									return c.get();
+								}
```

Comment:
Do we need two database look-ups here? Maybe we can look into return a DTO that includes both the concept and the mapping type then change the logic around here: if we have a SAME-AS map use that, etc. as before?

Reply:
Thanks @ibacher .
updated